### ISSUE TITLE
Heap buffer overflow in Log Flush

### ISSUE THREAD

**Issue by scw00** [apache/trafficserver#5385]
```
=================================================================
==106949==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x60200000b480 at pc 0x7ffff6a0b0e6 bp 0x7fffeea6d440 sp 0x7fffeea6d430
READ of size 4 at 0x60200000b480 thread T4 ([LOG_FLUSH])
    #0 0x7ffff6a0b0e5 in EventNotify::signal() /root/trafficserver/src/tscore/EventNotify.cc:73
    #1 0x555555d49f08 in LogObject::_checkout_write(unsigned long*, unsigned long) /root/trafficserver/proxy/logging/LogObject.cc:409
    #2 0x555555a41a61 in LogObject::force_new_buffer() /root/trafficserver/proxy/logging/LogObject.h:250
    #3 0x555555d4cad6 in LogObject::check_buffer_expiration(long) /root/trafficserver/proxy/logging/LogObject.cc:762
    #4 0x555555d4f2cb in LogObjectManager::check_buffer_expiration(long) /root/trafficserver/proxy/logging/LogObject.cc:1067
    #5 0x555555ce714e in Log::periodic_tasks(long) /root/trafficserver/proxy/logging/Log.cc:234
    #6 0x555555cf1e8f in Log::flush_thread_main(void*) /root/trafficserver/proxy/logging/Log.cc:1356
    #7 0x555555cf3500 in LoggingFlushContinuation::mainEvent(int, void*) (/usr/local/trafficserver/bin/traffic_server+0x79f500)
    #8 0x5555559f6825 in Continuation::handleEvent(int, void*) /root/trafficserver/iocore/eventsystem/I_Continuation.h:190
    #9 0x5555560d75d7 in EThread::execute() /root/trafficserver/iocore/eventsystem/UnixEThread.cc:318
    #10 0x5555560d3c45 in spawn_thread_internal /root/trafficserver/iocore/eventsystem/Thread.cc:92
    #11 0x7ffff53116da in start_thread (/lib/x86_64-linux-gnu/libpthread.so.0+0x76da)
    #12 0x7ffff44f788e in __clone (/lib/x86_64-linux-gnu/libc.so.6+0x12188e)

0x60200000b480 is located 0 bytes to the right of 16-byte region [0x60200000b470,0x60200000b480)
allocated by thread T0 ([TS_MAIN]) here:
    #0 0x7ffff6ef9970 in operator new[](unsigned long) (/usr/lib/x86_64-linux-gnu/libasan.so.5+0xef970)
    #1 0x555555cef90d in Log::create_threads() /root/trafficserver/proxy/logging/Log.cc:1023
    #2 0x555555cef633 in Log::init_when_enabled() /root/trafficserver/proxy/logging/Log.cc:1007
    #3 0x555555cef275 in Log::init(int) /root/trafficserver/proxy/logging/Log.cc:980
    #4 0x555555a64869 in main traffic_server/traffic_server.cc:1886
    #5 0x7ffff43f7b96 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21b96)

Thread T4 ([LOG_FLUSH]) created by T0 ([TS_MAIN]) here:
    #0 0x7ffff6e54a8f in pthread_create (/usr/lib/x86_64-linux-gnu/libasan.so.5+0x4aa8f)
    #1 0x5555560d34fd in ink_thread_create ../../include/tscore/ink_thread.h:159
    #2 0x5555560d3d75 in Thread::start(char const*, void*, unsigned long, std::function<void ()> const&) /root/trafficserver/iocore/eventsystem/Thread.cc:109
    #3 0x5555560dd5b6 in EventProcessor::spawn_thread(Continuation*, char const*, unsigned long) /root/trafficserver/iocore/eventsystem/UnixEventProcessor.cc:491
    #4 0x555555cefb62 in Log::create_threads() /root/trafficserver/proxy/logging/Log.cc:1049
    #5 0x555555cef633 in Log::init_when_enabled() /root/trafficserver/proxy/logging/Log.cc:1007
    #6 0x555555cef275 in Log::init(int) /root/trafficserver/proxy/logging/Log.cc:980
    #7 0x555555a64869 in main traffic_server/traffic_server.cc:1886
    #8 0x7ffff43f7b96 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21b96)

SUMMARY: AddressSanitizer: heap-buffer-overflow /root/trafficserver/src/tscore/EventNotify.cc:73 in EventNotify::signal()

```

**Comment by scw00** [comment]
```
Thread 1 "traffic_server" hit Breakpoint 1, LogObject::LogObject (this=0x60e00000abc0, format=0x7fffffffb990, log_dir=0x604000041290 "/usr/local/trafficserver/var/log/trafficserver",
    basename=0x7fffffffc640 "squid", file_format=LOG_FILE_BINARY, header=0x7fffffffc5c0 "", rolling_enabled=Log::ROLL_ON_TIME_ONLY, flush_threads=86400, rolling_interval_sec=0, rolling_offset_hr=10,
    rolling_size_mb=0, auto_created=false) at LogObject.cc:91
91	LogObject::LogObject(const LogFormat *format, const char *log_dir, const char *basename, LogFileFormat file_format,

```

**Comment by scw00** [comment]
@randall Could you please have look with that. It seems the LogObject params are mismatched. and it might relate to https://github.com/apache/trafficserver/commit/82d564486e058ead160bfb8ad15e40b0507ff8aa.

**Comment by randall** [comment]
@scw00 I think this fixes it https://github.com/apache/trafficserver/pull/5390

**Comment by scw00** [comment]
closes via #5390
